src/components/app.tsx

```diff
@@ -101,23 +103,18 @@ export default function App(props: AppProps): JSX.Element {
   const trajManager = React.useRef<RobotTrajectoryManager | undefined>(undefined);
 
   const doorStateManager = React.useMemo(() => new DoorStateManager(), []);
-  const [doorStates, setDoorStates] = React.useState<Readonly<Record<string, RomiCore.DoorState>>>(
-    {},
-  );
+  const [doorStates, setDoorStates] = React.useState(doorStateManager.doorStates());
```

Comment:
```suggestion
  const [doorStates, setDoorStates] = React.useState(() => doorStateManager.doorStates());
```
`doorStateManager.doorStates()` creates a shallow copy of all the current states, which may be somewhat expensive so its better to use lazy initial state.

Reply:
Yeah! you're totally right! Thank you!